backend/src/decorators/auth.js

```diff
@@ -16,11 +16,48 @@ const fp = require('fastify-plugin');
 const AuthRepo = require('../services/auth/auth.repo');
 const AuthPermissions = require('../services/auth/auth.permissions');
 
-module.exports = fp(async function authDecorators(fastify, opts) {
+// eslint-disable-next-line no-unused-vars
+module.exports = fp(async function authDecorators(fastify, _opts) {
   const authRepo = new AuthRepo(fastify.db);
   const authPermissions = new AuthPermissions(authRepo);
 
   fastify.decorate('authenticate', async function (req, reply) {
+    // Test mode: bypass authentication in development
+    const isTestMode =
+      process.env.NODE_ENV === 'development' &&
+      process.env.TEST_MODE === 'true';
+
+    if (isTestMode) {
+      try {
+        // Use a test user from the database (first user in seed data)
+        const testUser = await fastify.db.users.findFirst({
+          where: { deleted_at: null },
+          orderBy: { id: 'asc' },
+        });
```

Comment:
[Suggestion]
Instead of just getting the first user, we can pass in a custom header (say, `x-test-user-id` or `x-test-user-email`) to fetch a desired user. This would allow us to test the endpoints with users having different global and course roles.